### ISSUE TITLE
oxfmt: SCSS interpolation + a `key: value` argument in an unknown at-rule prelude fails to parse — regression in v0.57.0

### ISSUE THREAD

**Issue by joelfillmore** [oxc-project/oxc#25272]
### Input

```scss
/* repro.scss */
@insert-tokens theme: '#{$t}';
```

`@insert-tokens` is a custom at-rule consumed by a PostCSS plugin after Sass compiles. Any unknown at-rule reproduces it — the name is not significant.

### Config

```jsonc
{}
```

Config-independent — it also reproduces with no `.oxfmtrc.json` present at all.

### Oxfmt output

Oxfmt version: `0.62.0`

```
  x Syntax error: expect token `;`, but found `:`
   ,-[repro.scss:1:21]
 1 | @insert-tokens theme: '#{$t}';
   :                     ^
   `----
Error occurred when checking code style in the above files.
```

The file is rejected outright, so nothing is formatted. Broken in every version from 0.57.0 through 0.62.0 (current); 0.56.0 is fine.

### Prettier output

Prettier version: `3.9.4` (CLI), `3.9.6` (the linked playground)

```scss
/* repro.scss */
@insert-tokens theme: '#{$t}';
```

Output is byte-identical to the input — Prettier parses it and changes nothing (`All matched files use Prettier code style!`). oxfmt 0.56.0, which used the Prettier fallback, also accepted it. dart-sass 1.100.0 compiles it without complaint.

### Prettier playground link

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVAAgE5wA7YQB0AzmCSZuqgDpQACAllCXNjALQwQDWClMABZwAtnCSYA5AGJgAEhgBfSQG4QAGhAQ8MRtBLJQAQ2yEA7gAUT-ZCCMAbM0YCeBzQCNsRsHxgBlIzEAGWY4ZAAzB1ZNCHcAKzgwGAB1LzxbAjhWbAA3MI8vHzh-PG9mAHNkGGwAVzhNVhFGKtr6kDgADzw2RjFYBwB5bq9ubAsIEkZdaFsEABMNdq6evpgHABU2KBNGLIiotsmocvs4AEUaiHh9+2iQOJIOvwrTi6uwpEjbtoBHS-gLIQ8AYUEYSBwoHA4HNoYtqkZGPYKgBhCAiERGWwOeyLI4nOAAQRg1UY7hqALYIUhNzughgInsyUEUyypTAcD8-CmjByU2ctnIbhAOTqAEkoDDYH4wNhGDoCRK-DBnKcaW0CBM4KkjOkUJlsnlFvMBuEqR8QPZwotmNkYICjOUMWrNKVsNlbGQKIsCMwUow5kJkAAOAAMmlwf0YuHtjsxnwOmjW7mS-sDSAATJoaqx1kZ3CCvndRO5oTC5kEjMcag64AAxCDYDHEipY8kQECKRRAA

### It's the combination, not either half

SCSS interpolation alone is fine, and a `key: value` argument alone is fine. Only together do they fail:

| Snippet | 0.62.0 |
| --- | --- |
| `@insert-tokens theme: 'y';` — `key: value`, no interpolation | ok |
| `@insert-tokens '#{$t}';` — interpolation, no `key: value` | ok |
| `@insert-tokens theme: '#{$t}';` — both | **fail** |
| `@insert-tokens theme: #{$t};` — bare interpolation form | **fail** |
| `@insert-tokens a: '#{$t}', b: '#{$u}';` | **fail** |
| `@insert-tokens a: b('#{$t}');` | **fail** |
| `@insert-tokens a: #fff;` / `a: '#fff';` — `#` but no interpolation | ok |

### Root cause

`parse_unknown_at_rule_prelude` ([`crates/oxc_css_parser/src/parser/at_rule/mod.rs`](https://github.com/oxc-project/oxc-css-parser/blob/main/crates/oxc_css_parser/src/parser/at_rule/mod.rs)) first tries a permissive raw-token path that accepts any token — including `:` — up to `{` / `;`. That path deliberately bails on interpolation:

```rust
Token::StrTemplate(..) | Token::HashLBrace(..) => {
    return Err(Error { kind: ErrorKind::TryParseError, span: parser.cursor.bump()?.span });
}
```

It then falls back to parsing the prelude as a structured Sass value list:

```rust
Syntax::Scss | Syntax::Sass => self.parse_maybe_sass_list(/* allow_comma */ true)?,
```

But `parse_maybe_sass_list` (`parser/sass.rs`) breaks on `Token::Colon`:

```rust
Token::LBrace(..) | Token::RBrace(..) | Token::RParen(..)
| Token::Semicolon(..) | Token::Colon(..) | Token::DotDotDot(..)
| Token::Eof(..) => break,
```

So the `:` is left unconsumed and the caller, expecting `;` or `{`, reports ``expect token `;`, but found `:` ``.

That explains the whole matrix: interpolation is what forces the prelude off the permissive raw path onto the value-expression path, and that path has no production for a bare `key: value` argument. `#fff` is a hash token rather than interpolation, which is why it stays on the raw path.

### A second, more confusing error message

Same root cause, but when the at-rule sits inside an interpolated selector within a mixin the error is misreported at the enclosing selector, a line *above* the actual at-rule:

```scss
@mixin m($name) {
    #{$sel} {
        @insert-tokens theme: '#{$name}', scope: 'theme';
    }
}
```

```
  x Syntax error: expect token `:`, but found `{`
   ,-[repro2.scss:2:13]
 1 | @mixin m($name) {
 2 |     #{$sel} {
   :             ^
 3 |         @insert-tokens theme: '#{$name}', scope: 'theme';
   `----
```

### Not a duplicate of #24726 or #25075

GitHub's similarity check flags #24726, and #25075 is genuinely close, so to be explicit — both are fixed, and I verified their own repros pass on 0.62.0:

- **#24726** (`:host-context(.parent .child)`) → `All matched files use the correct format.` on 0.62.0. That was a **selector** parsing bug, fixed by oxc-css-parser#130. Different code path (`parser/selector.rs`), different grammar.
- **#25075** (`@container row #{$bp} { … }`) → also passes on 0.62.0. That is the closest relative: interpolation in an at-rule prelude. **Bare interpolation in a prelude now works; interpolation combined with a `key: value` argument still does not.** That remaining gap is what this issue is about.

The three share a family resemblance — SCSS, v0.57.0 regression, native `oxc_formatter_css` — but they are distinct bugs.

### Why this should parse

- Sass compiles all of these. The prelude of an unknown at-rule is an interpolated value; `key: value` pairs in it are just tokens Sass passes through.
- Prettier (postcss-scss) parses and formats them unchanged, so this is a Prettier-compatibility difference.
- `key: value` arguments in a custom at-rule prelude are the normal way to pass named options to a PostCSS plugin that consumes the at-rule after Sass compiles — Sass resolves the interpolation first, so the plugin only ever sees literal values.
- A hard parse error blocks formatting of the whole file, and only the first occurrence per file is reported — so a user fixes one and immediately hits the next. In our repo, two files held four occurrences, and only one surfaced at a time.

### Environment

- oxfmt 0.57.0 through 0.62.0 broken, 0.56.0 ok
- macOS (darwin arm64), `oxfmt --check` over an SCSS design-token package
- No Oxfmt playground link: https://playground.oxc.rs rejects `scss` with "Unknown file extension: Please provide a valid file extension", so this can only be reproduced against the CLI.
